src/Client.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Utopia\Fetch;
+
+/**
+ * Client class
+ * @package Utopia\Fetch
+ */
+class Client
+{
+    public const METHOD_GET = 'GET';
+    public const METHOD_POST = 'POST';
+    public const METHOD_PUT = 'PUT';
+    public const METHOD_PATCH = 'PATCH';
+    public const METHOD_DELETE = 'DELETE';
+    public const METHOD_HEAD = 'HEAD';
+    public const METHOD_OPTIONS = 'OPTIONS';
+    public const METHOD_CONNECT = 'CONNECT';
+    public const METHOD_TRACE = 'TRACE';
+
+    public const CONTENT_TYPE_APPLICATION_JSON = 'application/json';
+    public const CONTENT_TYPE_APPLICATION_FORM_URLENCODED = 'application/x-www-form-urlencoded';
+    public const CONTENT_TYPE_MULTIPART_FORM_DATA = 'multipart/form-data';
+    public const CONTENT_TYPE_GRAPHQL = 'application/graphql';
+
+    /**
+     * Flatten request body array to PHP multiple format
+     *
+     * @param array<mixed> $data
+     * @param string $prefix
+     * @return array<mixed>
+     */
+    private function flatten(array $data, string $prefix = ''): array
+    {
+        $output = [];
+        foreach ($data as $key => $value) {
+            $finalKey = $prefix ? "{$prefix}[{$key}]" : $key;
+
+            if (is_array($value)) {
+                $output += $this->flatten($value, $finalKey); // @todo: handle name collision here if needed
+            } else {
+                $output[$finalKey] = $value;
+            }
+        }
+
+        return $output;
+    }
+    /**
+     * This method is used to make a request to the server
+     * @param string $url
+     * @param array<string, string> $headers
+     * @param string $method
+     * @param array<string>|array<string, mixed> $body
+     * @param array<string, mixed> $query
+     * @return Response
+     */
+    private function buildResponse(
+        string $url,
+        array $headers = [],
+        string $method = self::METHOD_GET,
+        array $body = [],
+        array $query = []
+    ): Response {
+        // Process the data before making the request
+        if(!$method) { // if method is not set, set it to GET by default
+            $method = self::METHOD_GET;
+        } else { // else convert the method to uppercase
+            if (!in_array($method, [self::METHOD_PATCH, self::METHOD_GET, self::METHOD_CONNECT, self::METHOD_DELETE, self::METHOD_POST, self::METHOD_HEAD, self::METHOD_OPTIONS, self::METHOD_PUT, self::METHOD_TRACE ])) {
+                throw new FetchException("Unsupported HTTP method");
+            } else {
+                $method = strtoupper($method);
+            }
+        }
+        if(!is_array($headers)) {
+            $headers = [];
+        }
```

Comment:
By type we know it will be array, and we already set default value in `array $headers = [],`. We can remove this check I believe.